sdk/core/AzureCore/Source/DataStructures/Collections.swift

```diff
@@ -180,15 +176,9 @@ public class PagedCollection<SingleElement: Codable>: PagedCollectionDelegate {
             return
         }
 
-        let delegate = self.delegate ?? self
-
         client.logger.info(String(format: "Fetching next page with: %@", continuationToken))
-        var queryParams = [QueryParameter]()
-        guard let url = delegate.continuationUrl(
-            continuationToken: continuationToken,
-            queryParams: &queryParams,
-            requestUrl: requestUrl
-        ) else { return }
+        guard let url = client.continuationUrl(forRequestUrl: requestUrl, withContinuationToken: continuationToken)
```

Comment:
Based on the discussion with TransferDelegate vs. closures, should we expose a closure option here as an option for those who would not want to implement what is essentially a delegate method?

Reply:
I don't particularly think so, since the protocol is internal rather than something the end consumer will be having to code up. So it really only affects library authors, and personally I think the protocol method is more intention-revealing than the closure. Plus if your client has no paginated APIs you can just not conform to `PageableClient` at all.